playbooks/roles/document_hosting_k8s/tasks/main.yaml

```diff
@@ -0,0 +1,207 @@
+                volumeMounts:
+                - name: nginx-config
+                  mountPath: /etc/nginx/conf.d/nginx-site.conf
+                  subPath: nginx-site.conf
+                - name: nginx-config
+                  mountPath: /usr/share/nginx/html/custom_404.html
+                  subPath: custom_404.html
+            volumes:
+              - configMap:
+                  name: "{{ document_hosting_cm_name }}"
+                  defaultMode: 420
+                name: nginx-config
+            affinity:
+              podAntiAffinity:
+                preferredDuringSchedulingIgnoredDuringExecution:
+                - weight: 100
+                  podAffinityTerm:
+                    labelSelector:
+                      matchExpressions:
+                      - key: "app.kubernetes.io/instance"
+                        operator: In
+                        values:
+                        - "{{ instance }}"
+                    topologyKey: kubernetes.io/hostname
+
+
+- name: Create Document Hosting Autoscaler
+  community.kubernetes.k8s:
+    context: "{{ context }}"
+    state: "{{ state }}"
+    namespace: "{{ namespace }}"
+    name: "{{ document_hosting_hpa_name }}"
+    api_version: "autoscaling/v2beta2"
+    kind: "HorizontalPodAutoscaler"
+    apply: "yes"
+    definition:
+      metadata:
+        labels:
+          app.kubernetes.io/name: "document-hosting"
+          app.kubernetes.io/instance: "{{ instance }}"
+          app.kubernetes.io/managed-by: "system-config"
+      spec:
+        scaleTargetRef:
+          kind: "Deployment"
+          name: "{{ document_hosting_dep_name }}"
+          apiVersion: "apps/v1"
+        minReplicas: 1
+        maxReplicas: 5
+        metrics:
+          - type: "Resource"
+            resource:
+              name: "cpu"
+              target:
+                type: "Utilization"
+                averageUtilization: 70
+
+
+- name: Create Document Hosting Service
+  community.kubernetes.k8s:
+    context: "{{ context }}"
+    state: "{{ state }}"
+    namespace: "{{ namespace }}"
+    name: "{{ document_hosting_svc_name }}"
+    kind: "Service"
+    api_version: "v1"
+    apply: "yes"
+    definition:
+      metadata:
+        labels:
+          app.kubernetes.io/name: "document-hosting"
+          app.kubernetes.io/instance: "{{ instance }}"
+          app.kubernetes.io/managed-by: "system-config"
+      spec:
+        type: "ClusterIP"
+        ports:
+          - name: http
+            port: 80
+            protocol: "TCP"
+            targetPort: 80
+        selector:
+          app.kubernetes.io/name: "document-hosting"
+          app.kubernetes.io/instance: "{{ instance }}"
+
+
+- name: Create Document Hosting Ingress
+  community.kubernetes.k8s:
+    context: "{{ context }}"
+    state: "{{ state }}"
+    namespace: "{{ namespace }}"
+    name: "{{ document_hosting_ing_name }}"
+    api_version: "networking.k8s.io/v1"
```

Comment:
even with cce 1.19 this didn't worked on graphite - ansible module just dies. Need to check that